crates/libs/bindgen/src/functions.rs

```diff
@@ -1,23 +1,30 @@
 use super::*;
+use std::collections::BTreeMap;
 
 pub fn gen_sys_functions(tree: &TypeTree, gen: &Gen) -> TokenStream {
     if gen.sys {
-        let mut tokens = quote! {};
+        // the keys in this map are the libraries from which the various functions
+        // are imported.  The corresponding values are non-empty TokenStreams that contain
+        // the declarations of the functions imported from those libraries.
+        let mut tokens_by_library: BTreeMap<String, TokenStream> = BTreeMap::new();
 
         for entry in tree.types.values() {
-            tokens.combine(&gen_function_if(entry, gen));
+            gen_function_if(&mut tokens_by_library, entry, gen);
         }
 
-        if !tokens.is_empty() {
-            quote! {
-                #[link(name = "windows")]
+        let mut tokens = quote! {};
+        // Because tokens_by_library is a BTreeMap, we're guaranteed to visit entries
+        // in ascending order by key.
+        for (library, lib_tokens) in tokens_by_library {
+            tokens.combine(&quote! {
+                #[cfg_attr(feature = "use_raw_dylib", link(name = #library, kind = "raw-dylib"))]
+                #[cfg_attr(not(feature = "use_raw_dylib"), link(name = "windows"))]
```

Comment:
Can we add a test for this? Perhaps a dylib version of this test:

https://github.com/microsoft/windows-rs/tree/master/crates/tests/sys